coderd/workspacebuilds.go

```diff
@@ -404,6 +404,16 @@ func (api *API) postWorkspaceBuilds(rw http.ResponseWriter, r *http.Request) {
 			ctx,
 			tx,
 			func(action policy.Action, object rbac.Objecter) bool {
+				if object.RBACObject().Type == rbac.ResourceWorkspace.Type && action == policy.ActionDelete {
+					workspaceObj, ok := object.(database.Workspace)
+					if ok {
+						prebuild := workspaceObj.PrebuildRBAC()
+						// Fallback to normal workspace auth check
+						if auth := api.Authorize(r, action, prebuild); auth {
+							return auth
+						}
+					}
+				}
```

Comment:
Suggestion: use inline conditional assignment e.g.

```
if workspaceObj, ok := object.(database.Workspace); ok {
  // prebuild auth
}
```

Reply:
Addressed in https://github.com/coder/coder/pull/18333/commits/b2c7bdd86ac4bff084037b23d6bf11a16cd040e4